mobile/src/employee-details/employee-details.tsx

```diff
@@ -64,6 +79,14 @@ export class EmployeeDetailsImpl extends Component<EmployeeDetailsProps & Employ
         const events = this.props.events;
         const nextEvents = nextProps.events;
 
+        if (!this.props.userEmployeePermissions.equals(nextProps.userEmployeePermissions)) {
+
+            const permissions = this.props.userEmployeePermissions.get(this.props.employee.employeeId);
+            const nextPermissions = nextProps.userEmployeePermissions.get(this.props.employee.employeeId);
```

Comment:
nextProps?

Reply:
you probably want to check employeeId equality first, before permissions

Reply:
I'm checking it in  `equals`

Reply:
I'm checking for only particular employee whether his permissions were changed. Since Profile uses userEmployeePermissions as well, this check prevents Profile's re-render